pyogrio/__init__.py

```diff
@@ -21,6 +21,9 @@
     read_bounds,
     read_info,
     set_gdal_config_options,
+    vsi_listtree,
+    vsi_rmtree,
+    vsi_unlink,
```

Comment:
Do we explicitly want to expose those functions publicly (top-level) to our users as well? (are there specific use cases we are thinking of?) 
Because if this is mostly for the tests, we can also import from `pyogrio.core` or `pyogrio.util`

Reply:
I struggled with this as well.  Some of them may be useful in order to leverage GDAL's VSI interface.  That said, if you are using a different package to create VSI resources (e.g., the `gdal` package used directly), maybe you should use that other package to handle any related cleanup.

If we do want to keep this public, want to put these in the top-level interface?  Maybe we should move the VSI related functions to `vsi.py` and expose these there, so that they are available, but compartmentalized for advanced usage a bit better?

Reply:
I think it is practical/logical when it is possible to create files in e.g. /vsimem/ that the functions to remove them again are available as well without having to install another packages...

Putting them in a seperate interface is obviously possible, but I don't really see the point. pyogrio doesn't have that many functions, and I was quite happy that the hierarchy was removed in shapely so introducing it here now...